plugins/modules/image_streamer_build_plan.py

```diff
@@ -0,0 +1,139 @@
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: image_streamer_build_plan
+short_description: Manages Image Stream OS Build Plan resources.
+description:
+    - "Provides an interface to manage Image Stream OS Build Plans. Can create, update, and remove."
+version_added: "2.9.0"
+requirements:
+    - "python >= 3.4.2"
+    - "hpeOneView >= 6.0.0"
+author: "Venkatesh Ravula (@VenkateshRavula)"
+options:
+    state:
+        description:
+            - Indicates the desired state for the OS Build Plan resource.
+              C(present) will ensure data properties are compliant with Synergy Image Streamer.
+              C(absent) will remove the resource from Synergy Image Streamer, if it exists.
+        choices: ['present', 'absent']
+        required: true
+        type: str
+    data:
+        description:
+            - List with OS Build Plan properties and its associated states.
+        required: true
+        type: dict
+extends_documentation_fragment:
+    - hpe.oneview.oneview
+    - hpe.oneview.oneview.params
+'''
+
+EXAMPLES = '''
+- name: Create an OS Build Plan
+  image_streamer_build_plan:
+    hostname: 172.16.101.48
+    username: administrator
+    password: my_password
+    api_version: 600
+    state: present
+    data:
+      name: 'Demo OS Build Plan'
+      description: "oebuildplan"
+      oeBuildPlanType: "deploy"
+  delegate_to: localhost
+- name: Update the OS Build Plan description and name
+  image_streamer_build_plan:
+    hostname: 172.16.101.48
+    username: administrator
+    password: my_password
+    api_version: 600
+    state: present
+    data:
+      name: 'Demo OS Build Plan'
+      description: "New description"
+      newName: 'OS Build Plan Renamed'
+  delegate_to: localhost
+- name: Remove an OS Build Plan
+  image_streamer_build_plan:
+    hostname: 172.16.101.48
+    username: administrator
+    password: my_password
+    api_version: 600
+    state: absent
+    data:
+        name: 'Demo OS Build Plan'
+  delegate_to: localhost
+'''
+
+RETURN = '''
+build_plan:
+    description: Has the OneView facts about the OS Build Plan.
+    returned: On state 'present'.
+    type: dict
+'''
+
+from ansible_collections.hpe.oneview.plugins.module_utils.oneview import OneViewModuleBase, OneViewModuleValueError
+
+
+class BuildPlanModule(OneViewModuleBase):
+    MSG_CREATED = 'OS Build Plan created successfully.'
+    MSG_UPDATED = 'OS Build Plan updated successfully.'
+    MSG_ALREADY_PRESENT = 'OS Build Plan is already present.'
+    MSG_DELETED = 'OS Build Plan deleted successfully.'
+    MSG_ALREADY_ABSENT = 'OS Build Plan is already absent.'
+
+    argument_spec = dict(
+        state=dict(
+            required=True,
+            choices=['present', 'absent']
+        ),
+        data=dict(required=True, type='dict')
+    )
+
+    def __init__(self):
+        super(BuildPlanModule, self).__init__(additional_arg_spec=self.argument_spec)
```

Comment:
Remove arguments from super method and follow the same all resources

Reply:
Done